lua/toggleterm.lua

```diff
@@ -352,22 +352,23 @@ local function setup_autocommands(_)
     end
 
     -- Extra mouse fix for tmux
-    -- If tmux mouse mode is enabled
-    local output = vim.fn.system 'tmux display -p "#{mouse}"'
-    if output:sub(1, 1) == "1" then
-      -- Disable tmux mouse while using toggleterm
-      api.nvim_create_autocmd({ "TermEnter", "WinEnter <buffer>" }, {
-        desc = "Disable tmux mouse while using toggleterm",
-        group = AUGROUP,
-        callback = function() vim.fn.system "tmux set mouse off" end,
-      })
-
-      -- Enable tmux mouse when mouse leaves toggleterm
-      api.nvim_create_autocmd({ "WinLeave <buffer>" }, {
-        desc = "Enable tmux mouse when mouse leaves toggleterm",
-        group = AUGROUP,
-        callback = function() vim.fn.system "tmux set mouse on" end,
-      })
+    if os.getenv "TMUX" ~= nil then
+      local output = vim.fn.system 'tmux display -p "#{mouse}"'
```

Comment:
Would be good to have a comment on what this command is doing? Also if it's slow it might be better to use `vim.fn.jobstart`

Reply:
that command prints 1 if mouse support is enabled in tmux. We are saying "If we are running tmux, and mouse support is enabled, then create this autocmds".

That's why we need it to be blocking. It is fast though, and it doesn't cause issues if the result is not the expected.

This patch is already been [merged in rnvimr](https://github.com/kevinhwang91/rnvimr/pull/150) in case you wanna check out. I just changed the names to match toggleterm.